tests/lava/proc/dense/test_models.py

```diff
@@ -0,0 +1,553 @@
+        shape = (3, 4)
+        num_steps = 6
+        # Set up external input to emulate every neuron spiking once on
+        # timestep t=4.
+        vec_to_send = np.ones((shape[1],), dtype=np.float)
+        send_at_times = np.repeat(False, (num_steps,))
+        send_at_times[3] = True
+        sps = VecSendandRecvProcess(shape=(shape[1],), num_steps=num_steps,
+                                    vec_to_send=vec_to_send,
+                                    send_at_times=send_at_times)
+        # Set up a Dense Process where a single input layer neuron projects to
+        # all output layer neurons.
+        weights = np.zeros(shape, dtype=np.float)
+        weights[:, 2] = [3, 4, 5]
+        dense = Dense(shape=shape, weights=weights)
+        # Receive neuron spikes
+        spr = VecRecvProcess(shape=(num_steps, shape[0]))
+        sps.s_out.connect(dense.s_in)
+        dense.a_out.connect(spr.s_in)
+        # Configure execution and run
+        rcnd = RunSteps(num_steps=num_steps)
+        rcfg = DenseRunConfig(select_tag='floating_pt')
+        dense.run(condition=rcnd, run_cfg=rcfg)
+        # Gather spike data and stop
+        spk_data_through_run = spr.spk_data.get()
+        dense.stop()
+        # Gold standard for the test
+        # Expected behavior is that a_out corresponding to output
+        # neurons 1-3 will be equal to 3, 4, and 5, respectively, at timestep 5.
+        expected_spk_data = np.zeros((num_steps, shape[0]))
+        expected_spk_data[4, :] = [3, 4, 5]
+        self.assertTrue(np.all(expected_spk_data == spk_data_through_run))
+
+    def test_float_pm_recurrence(self):
+        """
+         Tests that floating Dense ProcessModel has non-blocking dynamics for
+         recurrent connectivity architectures.
+         """
+        shape = (3, 3)
+        num_steps = 6
+        # Set up external input to emulate every neuron spiking once on
+        # timestep 4.
+        vec_to_send = np.ones((shape[1],), dtype=np.float)
+        send_at_times = np.repeat(True, (num_steps,))
+        sps = VecSendandRecvProcess(shape=(shape[1],), num_steps=num_steps,
+                                    vec_to_send=vec_to_send,
+                                    send_at_times=send_at_times)
+        # Set up Dense Process with fully connected recurrent connectivity
+        # architecture
+        weights = np.ones(shape, dtype=np.float)
+        dense = Dense(shape=shape, weights=weights)
+        # Receive neuron spikes
+        sps.s_out.connect(dense.s_in)
+        dense.a_out.connect(sps.a_in)
+        # Configure execution and run
+        rcnd = RunSteps(num_steps=num_steps)
+        rcfg = DenseRunConfig(select_tag='floating_pt')
+        dense.run(condition=rcnd, run_cfg=rcfg)
+        dense.stop()
+
+
+class TestDenseProcessModelFixed(unittest.TestCase):
+    """Tests for fixed-point, ProcessModels of Dense, which are bit-accurate
+    with Loihi hardware"""
+
+    def test_bitacc_pm_fan_out_e(self):
```

Comment:
Changed the test name to include the full word 'excitatory'. 

The `bit_accurate_loihi` tag and `fixed_pt` tag are not exactly the same because every ProcModel that is bit accurate with Loihi has fixed point precision, but not every ProcModel with fixed point precision is bit accurate with Loihi. (Conv is one such example of a ProcModel that is currently fixed point precision but not bit accurate with Loihi.) 

In the case of Dense, either the 'fixed_pt' tag or the 'bit_accurate_loihi' tag will tell the RunConfig to choose the PyDenseModelBitAcc ProcModel as Dense's implementation.